src/main/javascript/src/views/participation/distribution/ParticipationManualDistribution.vue

```diff
@@ -0,0 +1,210 @@
+        "Unassigned",
+      ];
+    },
+
+    arrayData: {
+      get: function() {
+        console.log("In Get");
+        const newArray = [];
+        for (let i = 0; i < this.conditions.length; i++) {
+          newArray.push([]);
+        }
+        // All the participant will go to 'Unparticipate' section
+        newArray.push(this.participants);
+        return newArray;
+      },
+      set: function(newValue) {
+        this.arrayDataProxy = newValue;
+      },
+    },
+  },
+  methods: {
+    ...mapActions({
+      fetchExposures: "exposures/fetchExposures",
+    }),
+
+    getExposure() {
+      this.fetchExposures(this.experiment.experimentId);
+    },
+
+    submitDistribution() {
+      const requestBody = [];
+      const conditionGroupIDMap = {};
+
+      this.getExposure();
+      const firstExposureId = this.exposures
+        .map((expo) => expo.exposureId)
+        .sort((a, b) => a - b)[0];
+
+      const firstExposure = this.exposures.filter(
+        (expo) => expo.exposureId === firstExposureId
+      )[0];
+
+      firstExposure.groupConditionList.map(
+        ({ groupId }, index) => (conditionGroupIDMap[index] = groupId)
+      );
+
+      this.arrayDataProxy.map((arrData, index) =>
+        arrData.map((participant) => {
+          if (conditionGroupIDMap[index] !== undefined) {
+            const temp = {
+              participantId: participant.participantId,
+              consent: participant.consent,
+              dropped: participant.dropped,
+              groupId: conditionGroupIDMap[index],
+            };
+            requestBody.push(temp);
+          }
+        })
+      );
+
+      participantService
+        .updateParticipants(this.experiment.experimentId, requestBody)
+        .then((response) => {
+          if (response?.status === 200) {
+            this.$router.push({
+              name: "ParticipationSummary",
+              params: { experiment: this.experiment.experimentId },
+            });
+          } else {
+            alert(response.error);
+          }
+        })
+        .catch((response) => {
+          console.log("submitParticipants | catch", { response });
+        });
+    },
+
+    moveToHandler(option, tempSelected) {
+      const selectedParticipantIDs = tempSelected.map(
+        (participant) => participant.participantId
+      );
+
+      const filteredParticipants = this.arrayDataProxy.map(
+        (conditionParticipantMap) =>
+          conditionParticipantMap.filter(
+            (participant) =>
+              !selectedParticipantIDs.includes(participant.participantId)
+          )
+      );
+
+      const idx = this.getConditionNames.indexOf(option);
+      filteredParticipants[idx] = [
+        ...filteredParticipants[idx],
+        ...tempSelected,
+      ];
+
+      this.arrayData = filteredParticipants;
+    },
+  },
+  beforeRouteEnter(to, from, next) {
+    // don't load new data after participant selection screen
```

Comment:
I don't think these comments are descriptive of the following code.

Reply:
Updating